src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,59 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+#ifdef ESCARGOT_ENABLE_ES2015
+static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(thisObject, RegExp, match);
+
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    RegExpObject* rx;
+
+    if (thisObject->isRegExpObject(state) == true) {
+        rx = thisObject->asRegExpObject(state);
+    } else {
+        rx = new RegExpObject(state, Value(thisObject).toString(state), String::emptyString);
+    }
+    //21.2.5.6.8
+    bool global = rx->option() & RegExpObject::Option::Global;
+    if (global == false) {
+        return builtinRegExpExec(state, rx, 1, &argv[0], true);
+    } else {
+        bool fullUnicode = rx->option() & RegExpObject::Option::Unicode;
+        rx->setLastIndex(state, Value(0));
+        ArrayObject* A = new ArrayObject(state);
+        size_t n = 0;
+
+        //21.2.5.6.8.g.i
+        while (true) {
+            //21.2.5.6.8.g.i
+            Value result = builtinRegExpExec(state, rx, 1, &argv[0], true);
+
+            //21.2.5.6.8.g.iii
+            if (result.isNull() == true) {
+                if (n == 0) {
+                    return Value(Value::Null);
+                } else {
+                    return A;
+                }
+            } else {
+                //21.2.5.6.8.g.iv
+                Value matchStr = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, Value(0));
+                A->defineOwnProperty(state, ObjectPropertyName(state, Value(n)), ObjectPropertyDescriptor(Value(matchStr), (ObjectPropertyDescriptor::PresentAttribute)(ObjectPropertyDescriptor::WritablePresent | ObjectPropertyDescriptor::NonEnumerablePresent | ObjectPropertyDescriptor::NonConfigurablePresent)));
```

Comment:
This is not correct. According to the standard  21.2.5.6.8.g.iv.3, you must create a data property here, which attributes are the following: `PropertyDescriptor{[[Value]]: V, [[Writable]]: true, [[Enumerable]]: true, [[Configurable]]: true}.`